tools/purge_from_cache.cmd

```diff
@@ -0,0 +1,162 @@
+@echo OFF
+pushd %~dp0
+
+:defaults
+:: default values
+set cache_dir=
+set dryrun=1
+set mipmap_ext=jpg
+set action=echo found stale mipmap: 
+call %~dp0\common.cmd
+
+:init
+:: remember the command line to show it in the end when not purging
+set "commandline=%0 %*"
+
+:arguments
+:: handle command line arguments
+for %%o in (%*) DO (
+  IF /I "%%~o"=="-h"            call :USAGE && goto :end
+  IF /I "%%~o"=="-help"         call :USAGE && goto :end
+  IF /I "%%~o"=="--help"        call :USAGE && goto :end
+  IF /I "%%~o"=="/?"            call :USAGE && goto :end
+    
+  IF /I "%%~o"=="-l"            call set "library=%%~2" && shift /1
+  IF /I "%%~o"=="--library"     call set "library=%%~2" && shift /1
+    
+  IF /I "%%~o"=="-c"            call set "cache_base=%%~2" && shift /1
+  IF /I "%%~o"=="--cache_base"  call set "cache_base=%%~2" && shift /1
+  
+  IF /I "%%~o"=="-C"            call set "cache_dir=%%~2" && shift /1
+  IF /I "%%~o"=="--cache_dir"   call set "cache_dir=%%~2" && shift /1
+    
+  IF /I "%%~o"=="-d"            call set "configdir=%%~2" && shift /1
+  IF /I "%%~o"=="--configdir"   call set "configdir=%%~2" && shift /1
+    
+  IF /I "%%~o"=="-p"            set dryrun=0
+  IF /I "%%~o"=="--purge"       set dryrun=0
+
+  shift /1
+)
+
+:prechecks
+tasklist \FI "IMAGENAME eq darktable.exe" \FI "STATUS eq running" >NUL 2>&1 && echo error: darktable is running, please exit first && exit /b 1
+:: get sqlite3.exe here: https://www.sqlite.org/2017/sqlite-tools-win32-x86-3170000.zip
+where sqlite3 >NUL 2>&1 || echo error: sqlite3.exe is not found, please add it somewhere in your PATH first && exit /b 1
+:: get sqlite3.exe here: https://frippery.org/files/busybox/busybox.exe
+where busybox >NUL 2>&1 || echo error: busybox.exe is not found, please add it somewhere in your PATH first && exit /b 1
+
+:: set the command to run for each stale file
+if %dryrun% EQU 0 set action=del /f /q
+
+:: test configdir:
+if NOT EXIST "%configdir%" echo error: configdir "%configdir%" doesn't exist && exit /b 1
+
+:: if you force configdir but not library:
+if NOT EXIST "%library%" set library=%configdir%\library.db
+
+:: test library:
+if NOT EXIST "%library%" echo error: library db "%library%" doesn't exist && exit /b 1
+
+:: the mipmap directory matching the selected library as defined on Linux:
+:: echo %LOCALAPPDATA%\darktable\library.db | busybox sha1sum
+:: 2917b0700b9fbb133931706610039dffd5f2d65d
+:: sha1sum calculation is disabled on Windows until I know how to generate it properly
+REM IF NOT DEFINED cache_dir (
+  REM for /f %%S in ('echo %library% ^| busybox sha1sum') DO set sha1sum=%%S
+  REM call set cache_dir=%cache_base%\mipmaps-%%sha1sum%%.d
+REM )
+
+:: autodetect cache_dir instead; there is 0.00001% chance that Windows users know how to handle multiple libraries, let alone cache folders
+set cache_dir_nb=0
+for /f %%c in ('dir /b /AD "%cache_base%\mipmaps-*.d" 2^>NUL') DO (set /A cache_dir_nb+=1 && set "cache_dir=%cache_base%\%%~c")
```

Comment:
I don't think this will work correctly if the user is running multiple versions of darktable, since there would be multiple caches.